### PR TITLE
Adds transform and atransform support to runnable sequences

### PR DESCRIPTION
**Pull request by jacoblee93**
Allow runnable sequences to support transform if each individual runnable inside supports transform/atransform.

@nfcampos

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
seems reasonable to me but will let @nfcampos tal

File: libs/langchain/langchain/schema/runnable/base.py [hunk 1]
```diff
@@ -1063,57 +1060,53 @@ def stream(
...
-        )
-        # start the root run
-        run_manager = await callback_manager.on_chain_start(dumpd(self), input)
-
         steps = [self.first] + self.middle + [self.last]
         streaming_start_index = len(steps) - 1
 
         for i in range(len(steps) - 1, 0, -1):
-            if type(steps[i]).transform != Runnable.transform:
+            if type(steps[i]).atransform != Runnable.atransform:
```
Comment (nfcampos) [inline libs/langchain/langchain/schema/runnable/base.py:1073]: I guess this was incorrect before?

File: libs/langchain/langchain/schema/runnable/base.py [hunk 2]
```diff
@@ -1139,6 +1132,47 @@ async def astream(
...
+    async def astream(
+        self,
+        input: Input,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> AsyncIterator[Output]:
+        async def input_aiter() -> AsyncIterator[Input]:
+            yield input
+
+        async for chunk in self.atransform(input_aiter(), config):
```
Comment (nfcampos) [inline libs/langchain/langchain/schema/runnable/base.py:1173]: Pass kwargs here for consistency

File: libs/langchain/langchain/schema/runnable/base.py (hunk 1 above)
Comment (jacoblee93) [reply libs/langchain/langchain/schema/runnable/base.py:1019]: I think so? Wouldn't make sense to compare to the non-async version here

**Comment by jacoblee93** [comment]
Merged and updated! Any other comments?

**Comment by jacoblee93** [comment]
@nfcampos updated to take into account #9753 - can you have another look?

**Review by nfcampos — APPROVED** [review]

**Pull request merged**